modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -615,7 +615,7 @@ public void addListener(InvalidationListener invalidationListener) {
     @Override
     public void removeListener(InvalidationListener invalidationListener) {
         if (invalidationListener != null) {
```

Comment:
`Observable.removeListener(InvalidationListener)` is specified to reject `null` by throwing NPE.

Reply:
Alright, I'll file another ticket for that.  The general state of this class is quite poor, as it does not follow any of the contracts of the interfaces it implements (including `Collection`, `Set` and `ObservableSet`).  This is bad because even though this class is an implementation detail, it does get exposed as an `ObservableSet` via `Node#getPseudoClassStates`.  Things that it violates for example are:

     Node.getPseudoClassStates().retainAll(null);   // no exception

Or:
   
     Node.getPseudoClassStates().retainAll(set2);   // does nothing if set2 is not a BitSet

Or:

     // fails, even if they're the same because `equals` assumes another BitSet     
    Node.getPseudoClassStates().equals(Set.of(PseudoClass.getPseudoClass("armed"));

     // works:
     Set.of(PseudoClass.getPseudoClass("armed")).equals(Node.getPseudoClassStates());

Same goes for the hash code value; these are strictly defined for sets, but `BitSet` violates it causing problems when using these sets as keys.  This causes numerous odd problems that I had to track down while implementing #1076 -- I can only assume very few people are using these API's in the wild.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8305816 so we don't forget.